sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala

```diff
@@ -791,4 +791,42 @@ class JsonExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper with
         checkDecimalInfer(_, """struct<d:decimal(7,3)>""")
     }
   }
+
+  test("json_object_keys") {
+    val null_object = """"""
+    val empty_json_object = """{}"""
+    val simple_json_object = """{"key": 1}"""
+    val another_simple_json_object = """{"key": "value", "key2": 2}"""
+    val json_object_with_array = """{"arrayKey": [1, 2, 3]}"""
+    val another_json_object_with_array = """{"key":[1,2,3,{"key":"value"},[1,2,3]]}"""
+    val complex_json_object = """{"f1":"abc","f2":{"f3":"a", "f4":"b"}}"""
+    val another_complex_json_object = """{"k1": [1, 2, {"key": 5}], "k2": {"key2": [1, 2]}}"""
+    val empty_json_array = """[]"""
+    val invalid_json_object = """{[1,2]}"""
+    val another_invalid_json_object = """{"key": 45, "random_string"}"""
```

Comment:
Please use the simpler pattern which I commented as an example in your sister PR.